### PR TITLE
Connection error box style fix

### LINKED ISSUE #14344: Connection error box has extra border
- Azure Data Studio Version:
Screen Shot 2021-02-18 at 4 10 54 PM

Steps to Reproduce:

1. Try to connect to a DB that will throw a connection error (in my case, I did not have the docker container running for the DB I was trying to connect to)
2. The connection error box has this extra border under it
Screen Shot 2021-02-18 at 4 06 34 PM

### PR DESCRIPTION
**Pull request by halerankin**
Refactor of modal.ts, moves callout-specific style into a separate stylesheet: calloutDialog.css now imported by modal.ts. Adds new attachCalloutDialogStyler.
Modifies modal styles, allowing maximum height of basic modal to be 480px. 

This PR is work done for 14344

---alanren---
this PR fixes #14344

### CODE REVIEW

**Comment by alanrenmsft** [comment]
the issue is not correctly linked

**Review by alanrenmsft — CHANGES_REQUESTED** [review]
this change does make the extra border go away, but is not the root cause of the issue, as I mentioned in the issue

https://user-images.githubusercontent.com/13777222/109074027-05e0c080-76ac-11eb-8d35-acfe63d2691b.png

you introduced some colors that is not relevant to dialogs in the original PR, please revert those changes and if you do need border for callout, please implement a new attachCalloutDialogStyler to pass in the correct colors, the style method in the modal.ts shouldn't be changed.

File: src/sql/workbench/browser/modal/media/modal.css [hunk 1]
```diff
@@ -21,17 +21,13 @@
 .modal:not(.flyout-dialog):not(.callout-dialog) .modal-dialog {
 	margin: auto;
 	width: 640px;
-	height: 480px;
+	max-height: 480px;
```
Comment (alanrenmsft) [inline src/sql/workbench/browser/modal/media/modal.css:24]: with the 'correct' fix, you don't need this change anymore, right? let's keep the old behavior to avoid unexpected impacts.

File: src/sql/workbench/browser/modal/modal.ts [hunk 2]
```diff
@@ -459,7 +456,7 @@ export abstract class Modal extends Disposable implements IThemable {
 	 * Shows the modal and attaches key listeners
 	 */
 	protected show() {
-		this.positionDialog();
+		this.positionCalloutDialog();
```
Comment (alanrenmsft) [inline src/sql/workbench/browser/modal/modal.ts:459]: if this is only applicable to callout dialog, we should check the style before calling the method.

**Review by alanrenmsft — APPROVED** [review]

File: src/sql/workbench/contrib/notebook/browser/calloutDialog/media/calloutDialog.css [hunk 3]
```diff
@@ -49,7 +42,6 @@
 	transform: rotate(135deg);
 }
 .callout-arrow.from-left:before {
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/notebook/browser/calloutDialog/media/calloutDialog.css:44]: Would these be applicable to ALL callout dialogs or just the notebook specific ones? If it's all then they should really go in modal.css

File: src/sql/workbench/services/dialog/browser/dialogModal.ts [hunk 4]
```diff
@@ -56,7 +56,11 @@ export class DialogModal extends Modal {
 
 	public render() {
 		super.render();
-		attachModalDialogStyler(this, this._themeService);
+		if (this._modalOptions.dialogStyle === 'callout') {
```
Comment (Charles-Gagnon) [inline src/sql/workbench/services/dialog/browser/dialogModal.ts:59]: I feel like a helper function `attachModalDialogTypeStyler` or something would be good to have that would attach the correct styler based on the type of modal. If we add more types in the future it'd be easy to miss updating some of these

File: src/sql/workbench/browser/modal/modal.ts [hunk 5]
```diff
@@ -418,10 +414,10 @@ export abstract class Modal extends Disposable implements IThemable {
 
 
 	/**
-	 * Tasks to perform before dialog is shown
+	 * Tasks to perform before calout dialog is shown
 	 * Includes: positioning of dialog
 	 */
-	protected positionDialog(): void {
+	protected positionCalloutDialog(): void {
```
Comment (Charles-Gagnon) [inline src/sql/workbench/browser/modal/modal.ts:420]: Why did you change this? It's still positioning the normal modal dialog right?

**Review by Charles-Gagnon — CHANGES_REQUESTED** [review]

File: src/sql/workbench/browser/modal/media/modal.css (hunk 1 above)
Comment (halerankin) [reply src/sql/workbench/browser/modal/media/modal.css:24]: This fix will remain. To be sure, I tested the modal-body div, adding several paragraphs of information. Overflow causes content to scroll inside the modal. This is helpful.

File: src/sql/workbench/browser/modal/modal.ts (hunk 2 above)
Comment (halerankin) [reply src/sql/workbench/browser/modal/modal.ts:459]: Fixed.

File: src/sql/workbench/contrib/notebook/browser/calloutDialog/media/calloutDialog.css (hunk 3 above)
Comment (halerankin) [reply src/sql/workbench/contrib/notebook/browser/calloutDialog/media/calloutDialog.css:45]: The calloutDialog style is a concept that's current in play in notebook but certainly not limited to notebook. I would elect to keep these styles in calloutDialog.css. Modal and flyout type do not use this style. I did what I could to make these generic enough so that, when calling the calloutDialog, one can specify the dialog position with respect to the triggering element.

File: src/sql/workbench/browser/modal/modal.ts (hunk 5 above)
Comment (halerankin) [reply src/sql/workbench/browser/modal/modal.ts:420]: This is all my original work, exclusive to callout. I simply renamed it.

File: src/sql/workbench/services/dialog/browser/dialogModal.ts (hunk 4 above)
Comment (halerankin) [reply src/sql/workbench/services/dialog/browser/dialogModal.ts:59]: This feature has been in the works for months now -- do we want to implement this now or create a followup task?

Comment (Charles-Gagnon) [reply src/sql/workbench/services/dialog/browser/dialogModal.ts:59]: The ask here isn't very big - and considering that you're just now adding this special type of styler based on the type of modal dialog I don't think it's unreasonable to do now.

File: src/sql/workbench/browser/modal/modal.ts [hunk 6]
```diff
@@ -418,48 +414,49 @@ export abstract class Modal extends Disposable implements IThemable {
 
 
 	/**
-	 * Tasks to perform before dialog is shown
+	 * Tasks to perform before calout dialog is shown
```
Comment (Charles-Gagnon) [inline src/sql/workbench/browser/modal/modal.ts:417]: `callout`

File: src/sql/workbench/contrib/notebook/browser/calloutDialog/media/calloutDialog.css (hunk 3 above)
Comment (Charles-Gagnon) [reply src/sql/workbench/contrib/notebook/browser/calloutDialog/media/calloutDialog.css:45]: You can put them in the css but still keep the rule selectors as they are and it would be the same thing. All the styles get bundled into a huge single CSS anyways so it's not like it'd really be changing anything.

The point of moving it would be to keep the styles next to the thing that actually uses it. So since these apply to all callout dialogs they should go in modal.css (or if you prefer you can move this file over there and then also import this css in modal.ts)

File: src/sql/workbench/services/dialog/browser/dialogModal.ts (hunk 4 above)
Comment (halerankin) [reply src/sql/workbench/services/dialog/browser/dialogModal.ts:59]: Agreed. How/where should I begin?

**Review by Charles-Gagnon — APPROVED** [review]
Should be good once you fix the merge conflicts

File: src/sql/workbench/common/styler.ts [hunk 7]
```diff
@@ -49,3 +50,22 @@ export function attachTabbedPanelStyler(widget: IThemable, themeService: IThemeS
 		activeTabContrastBorder: cr.activeContrastBorder
 	}, widget);
 }
+
+export function attachCalloutDialogStyler(widget: IThemable, themeService: IThemeService, style?:
+	{
```
Comment (Charles-Gagnon) [inline src/sql/workbench/common/styler.ts:55]: Please make a type for this similar to https://github.com/microsoft/azuredatastudio/blob/main/src/vs/platform/theme/common/styler.ts#L52

File: src/sql/workbench/services/dialog/browser/dialogModal.ts (hunk 4 above)
Comment (halerankin) [reply src/sql/workbench/services/dialog/browser/dialogModal.ts]: Added conditional within styler to provide colors for dialogModal depending on dialogStyle

File: src/sql/workbench/common/styler.ts [hunk 8]
```diff
@@ -49,3 +55,34 @@ export function attachTabbedPanelStyler(widget: IThemable, themeService: IThemeS
...
+		dialogBodyBackground: (style && style.dialogBodyBackground) || sqlcr.calloutDialogBodyBackground,
+		dialogInteriorBorder: (style && style.dialogInteriorBorder) || sqlcr.calloutDialogInteriorBorder,
+		dialogExteriorBorder: (style && style.dialogExteriorBorder) || sqlcr.calloutDialogExteriorBorder,
+		dialogShadowColor: (style && style.dialogShadowColor) || sqlcr.calloutDialogShadowColor
+	} as IModalDialogStyleOverrides, widget);
+}
+
+export function attachCustomDialogStyler(widget: IThemable, themeService: IThemeService, dialogStyle?: string, style?: IModalDialogStyleOverrides): IDisposable {
+	if (dialogStyle === 'callout') {
+		return attachStyler(themeService, {
```
Comment (Charles-Gagnon) [inline src/sql/workbench/common/styler.ts:72]: Just call `attachCalloutDialogStyler` and `attachModalDialogStyler` here instead
